analyzers/src/SonarAnalyzer.CSharp/Rules/AssignmentInsideSubExpression.cs

```diff
@@ -72,11 +72,15 @@ public sealed class AssignmentInsideSubExpression : SonarDiagnosticAnalyzer
 
         private static bool IsNonCompliantSubExpression(AssignmentExpressionSyntax assignment, ExpressionSyntax topParenthesizedExpression) =>
             IsInsideExpression(topParenthesizedExpression)
+            && !IsInWithInitializerExpression(topParenthesizedExpression)
             && !IsCompliantAssignmentInsideExpression(assignment, topParenthesizedExpression);
 
         private static bool IsInsideExpression(ExpressionSyntax expression) =>
             expression.Parent.FirstAncestorOrSelf<ExpressionSyntax>() != null;
 
+        private static bool IsInWithInitializerExpression(ExpressionSyntax expression) =>
```

Comment:
With this implementation we have inconsistency as with ObjectInitializerExpression we have FNs and with WIthInitializerExpression we raise as expected. Also rename this method and use IsAnyKind with both InitializerExpressions. In addition if we remove ObjectInitializerExpression from the AllowedParentKinds we should be able to cover the following case: `var v7 = new MyClass { MyField = i = 42 }; // Noncompliant`